### PR TITLE
CSV Annotations can be imported as POI

### PR DESCRIPTION
**Pull request by peterAdams1014**
Added CSV Parser.
Added Import Annotations UI.
Fixes [AB#2131](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/2131)

### CODE REVIEW

**Comment by peterAdams1014** [comment]
I am unsure of the requirements surrounding the ID and ParentID columns. I have temporarily compiled that WIP feature out with the TEMP_DISABLE_PARENT_ID macro. This will be resolved in a later PR.

**Comment by peterAdams1014** [comment]
This adds a new modal:
image

File: src/parsers/vcCSV.cpp [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+  return result;
+}
+
+udResult vcCSV_ReadHeader(vcCSV *pCSV)
+{
+  udResult result;
+
+  while (!pCSV->completeRead)
+  {
+    const size_t minimumHeaderReadSize = CAPACITY_CHUNK_SIZE; // 8kb
```
Comment (samsinsane) [inline src/parsers/vcCSV.cpp]: You should put this comment with the `CAPACITY_CHUNK_SIZE`, also I think it's actually 16kb?

File: src/parsers/vcCSV.cpp [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+  pCSV->data.pData = udAllocType(char, CAPACITY_CHUNK_SIZE, udAF_Zero);
+  pCSV->data.capacityBytes = CAPACITY_CHUNK_SIZE;
+
+  result = udR_Success;
+  *ppCSV = pCSV;
+  pCSV = nullptr;
+epilogue:
+
+  if (pCSV != nullptr)
```
Comment (samsinsane) [inline src/parsers/vcCSV.cpp:271]: This isn't entirely necessary, it is performed by `vcCSV_Destroy`.

File: src/parsers/vcCSV.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#ifndef vcCSV_h__
+#define vcCSV_h__
+
+#include "udResult.h"
+#include "udChunkedArray.h"
+
+enum vcCSV_ColumnType
```
Comment (samsinsane) [inline src/parsers/vcCSV.h:7]: As mentioned in the other comment, this should be in a different file too.

**Review by samsinsane — DISMISSED** [review]

File: src/parsers/vcCSV.cpp [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+#define MAX_COLUMN_COUNT 16
+#define CAPACITY_CHUNK_SIZE (1024 << 4) 
+
+bool vcCSV_ShouldSkipLine(char *pToken)
+{
+  // Skip over comments etc
+  return pToken[0] == 0 || pToken[0] == ' ' || udStrBeginsWith(pToken, "//") || udStrBeginsWith(pToken, "!");
+}
+
+void vcCSV_InferColumnTypes(vcCSV *pCSV)
```
[the remark this answers is not in the document]
Comment (peterAdams1014) [reply src/parsers/vcCSV.cpp:16]: Agreed. I will make this change in a future PR.

**Review by samsinsane — DISMISSED** [review]

**Review by samsinsane — APPROVED** [review]

**Pull request merged**